OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd

```diff
@@ -0,0 +1,682 @@
+reagent batch differences. Appending the project name to the end of the
+control samples can ensure unique Sample IDs.
+
+```{r, echo=TRUE, eval=FALSE}
+data.frame(SampleID = intersect(data_E3072$SampleID, data_EHT$SampleID)) |>
+  dplyr::filter(!stringr::str_detect(SampleID, "CONTROL_SAMPLE"))
+```
+
+```{r, echo=FALSE, eval=FALSE}
+data.frame(SampleID = intersect(data_E3072$SampleID, data_EHT$SampleID)) |>
+  dplyr::filter(!stringr::str_detect(SampleID, "CONTROL_SAMPLE")) |> #Remove control samples
+  kableExtra::kbl(booktabs = TRUE,
+      digits = 2,
+      caption = "Table 3. Overlapping bridging samples") |>
+  kableExtra::kable_styling(bootstrap_options = "striped", full_width = FALSE,
+                position = "center", latex_options = "HOLD_position")
+```
+
+```{r, message=FALSE, fig.cap=f3, eval=FALSE}
+#### Extract bridging samples
+
+overlapping_samples <-  data.frame(SampleID = intersect(data_E3072$SampleID, data_EHT$SampleID)) |>  
+  filter(!str_detect(SampleID, "CONTROL_SAMPLE")) |> #Remove control samples
+  pull(SampleID)
+
+npx_before_br <- data_E3072 |>
+  dplyr::filter(!str_detect(SampleID, "CONTROL_SAMPLE")) |> #Remove control samples
+  dplyr::mutate(Type = if_else(SampleID %in% overlapping_samples,
+                        paste0("Explore 3072 Bridge"),
+                        paste0("Explore 3072 Sample"))) |>
+  rbind({
+    data_EHT |>
+      filter(!str_detect(SampleID, "CONTROL_SAMPLE")) |> #Remove control samples |> 
+      mutate(Type = if_else(SampleID %in% overlapping_samples,
+                            paste0("Explore HT Bridge"),
+                            paste0("Explore HT Sample"))) |>
+      mutate(SampleID = if_else(SampleID %in% overlapping_samples,
+                                paste0(SampleID, "_new"),
+                                SampleID))
+  })
+
+### PCA plot
+OlinkAnalyze::olink_pca_plot(df          = npx_before_br,
+                             color_g     = "Type")
+```
+
+## Normalization
+
+The `olink_normalization_bridge_product()` function is used to determine
+which assays are bridgeable and of the bridgeable assays what
+normalization method is advised, and to calculate normalized NPX values
+for the Explore 3072 (non-reference) project. Normalized NPX values are calculated for
+all assays across products as described in the Workflow Overview and in
+the sections below. Within this function the bridging recommendations
+for each assay are determined and the NPX values are normalized using
+the two methods described below.
+
+```{r, eval=FALSE}
+# Find shared samples
+npx_1 <- data_EHT |>
+  mutate(Project = "data1") 
+npx_2 <- data_E3072 |>
+  mutate(Project = "data2")
+
+overlap_samples <-data.frame(SampleID = intersect(npx_1$SampleID, npx_2$SampleID)) |>
+  filter(!str_detect(SampleID, "CONTROL_SAMPLE")) |> #Remove control samples
+  pull(SampleID)
+
+overlap_samples_list <- list("DF1" = overlap_samples,
+                             "DF2" = overlap_samples)
```

Comment:
TBD - will we have the olink_normalization_bridge_product() wrapper ready for initial launch? If not, should we just refer to olink_normalization()? Will leave as-is for now.